File: nodejs_services/lambda/local_test.js
/**
 * http://usejsdoc.org/
 */

var zip_file_url = {
	"100": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/100_sales_records.csv",
	"1000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/1000_sales_records.csv",
	"5000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/5000_sales_records.csv",
	"10000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/10000_sales_records.csv",
	"50000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/50000_sales_records.csv",
	"100000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/100000_sales_records.csv",
	"500000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/500000_sales_records.csv",
	"1000000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/1000000_sales_records.csv",
	"1500000": "http://cssgate.insttech.washington.edu/~varikmp/tcss562/1500000_sales_records.csv"
}

var request = {
		"csv_url_file": zip_file_url["100"],
		"csv_file_name": "100_sales_records.csv",
		"sales_channel": "Online",
		"order_priority": "Critical"
}


function main(params) {
	//return (require('./extract'))(params);
	return (require('./load'))(params);
	//return (require('./query'))(params);
}

console.log(main(request));

File: python_services/lambda/local_test.py
'''
Created on Nov 24, 2019

@author: varikmp
'''
from handler import extract, load, query

my_request = dict()
  
# http://faculty.washington.edu/wlloyd/courses/tcss562/project/etl/sales_data/

data_size = '100' # '100', '1000', '5000', '10000', '50000', '100000', '500000', '1000000', '1500000'
file_pattern = "{}_sales_records.csv"
my_request["csv_file_name"] = file_pattern.format(data_size)
csv_link = 'http://cssgate.insttech.washington.edu/~varikmp/tcss562/' + my_request["csv_file_name"]
my_request["csv_file_url"] = csv_link
my_request['overwrite_table'] = 1 # > 0 to overwrite the table

my_request['region'] = 'Asia'
my_request['sales_channel'] = 'Online'

# print(extract(my_request, True))
# print(load(my_request, True))
print(query(my_request, True))

# from utilities import is_bucket_existed, list_all_buckets
# print(list_all_buckets())
# print(is_bucket_existed('tcss562.mylogs.vmp'))

# from utilities import upload_to_bucket
# upload_to_bucket('/tmp/100_sales_records.csv')


# from utilities import download_if_file_exists_on_amazon_s3
# download_if_file_exists_on_amazon_s3('processed_100_sales_records.csv')

